engines/osxnative/engine.go

```diff
@@ -12,22 +12,40 @@ import (
 
 type engine struct {
 	engines.EngineBase
-	log *logrus.Entry
+	config *configType
+	log    *logrus.Entry
 }
 
 type engineProvider struct {
 	engines.EngineProviderBase
 }
 
 func (e engineProvider) NewEngine(options engines.EngineOptions) (engines.Engine, error) {
-	return engine{log: options.Log}, nil
+	var c configType
+	err := configSchema.Map(options.Config, &c)
+	if err == schematypes.ErrTypeMismatch {
+		panic("Type mismatch")
+	} else if err != nil {
+		options.Log.WithError(err).Error("Invalid configuration")
+		return nil, engines.ErrContractViolation
+	}
+
+	return engine{
```

Comment:
Giving engine data members, I would suggest you pass it as a pointer... to avoid copying for every method call... also if you were to mutate members it wouldn't work unless it's by reference...